src/servers/src/auth.rs

```diff
@@ -99,23 +100,31 @@ pub fn user_provider_from_option(opt: &String) -> Result<UserProviderRef, Error>
 #[snafu(visibility(pub))]
 pub enum Error {
     #[snafu(display("Invalid config value: {}, {}", value, msg))]
-    InvalidConfig {
-        value: String,
-        msg: String,
+    InvalidConfig { value: String, msg: String },
+
+    #[snafu(display("IO error, source: {}", source))]
+    Io {
+        source: std::io::Error,
         backtrace: Backtrace,
     },
 
-    #[snafu(display("Encounter IO error, source: {}", source))]
-    IOErr { source: std::io::Error },
+    #[snafu(display("Tonic status error, source: {}", source))]
+    TonicStatus {
```

Comment:
I think we can merge this into `AuthBackend` error.

Reply:
it seems `Error` should impl `ErrorExt` to be converted to `BoxedError`, both `std::io::Error` and `tonic::status` failed to meet the requirement.

Reply:
I see. Having `TonicStatus` here is a little weird. As an API definition, it assumes we have an RPC implementation, and even worse, it assumes the RPC is implemented by tonic library. I think a better solution is to put this error into an auth implementation and wrap it as `BoxedError` of `AuthBackend`

Reply:
good point. `TonicStatus` is removed now